src/manager/framework/api/framework_api.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/Dataman-Cloud/swan/src/apiserver"
+	"github.com/Dataman-Cloud/swan/src/apiserver/metrics"
+	"github.com/Dataman-Cloud/swan/src/config"
+	"github.com/Dataman-Cloud/swan/src/manager/framework/scheduler"
+	"github.com/Dataman-Cloud/swan/src/types"
+
+	"github.com/emicklei/go-restful"
+)
+
+type FrameworkService struct {
+	sched *scheduler.Scheduler
+	apiserver.ApiRegister
+}
+
+func NewAndInstallFrameworkService(apiServer *apiserver.ApiServer, sched *scheduler.Scheduler) *FrameworkService {
+	frameworkService := &FrameworkService{
+		sched: sched,
+	}
+	apiserver.Install(apiServer, frameworkService)
+	return frameworkService
+}
+
+func (fs *FrameworkService) Register(container *restful.Container) {
+	ws := new(restful.WebService)
+	ws.
+		ApiVersion(config.API_PREFIX).
+		Path(config.API_PREFIX + "/framework").
+		Doc("framework info").
+		Produces(restful.MIME_JSON)
+
+	ws.Route(ws.GET("/info").To(metrics.InstrumentRouteFunc("GET", "Info", fs.Info)).
+		Doc("Info").
+		Operation("info").
+		Returns(200, "OK", types.FrameworkInfo{}))
+
+	container.Add(ws)
+}
+
+func (fs *FrameworkService) Info(req *restful.Request, resp *restful.Response) {
+	info := &types.FrameworkInfo{
+		Id: fs.sched.MesosConnector.FrameworkInfo.GetId().GetValue(),
```

Comment:
check frameworkInfo nil, because there is not such info before subscribed got return.